packages/hash/integration/src/tests/integration.test.ts

```diff
@@ -1256,6 +1259,157 @@ describe("logged in user ", () => {
     });
   });
 
+  describe("can create page trees", () => {
+    let superPage: PageFieldsFragment;
+    const superTitle = "Super page 1";
+    beforeAll(async () => {
+      superPage = await client.createPage({
+        accountId: existingUser.accountId,
+        properties: {
+          title: superTitle,
+        },
+      });
+    });
+
+    let subPage: PageFieldsFragment;
+    it("can create a subpage", async () => {
+      // - Super page 1
+      //   - sub page 1
+      const title = "sub page 1";
+      subPage = await client.createPage({
+        accountId: existingUser.accountId,
+        properties: {
+          title,
+        },
+      });
+
+      const result = await client.setParentPage({
+        accountId: existingUser.accountId,
+        pageId: subPage.entityId,
+        parentPageId: superPage.entityId,
+      });
+
+      const pageTree = await client.getAccountPagesTree({
+        accountId: existingUser.accountId,
+      });
+
+      expect(result.entityId).toEqual(subPage.entityId);
+      expect(pageTree).toContainEqual({
+        entityId: subPage.entityId,
+        properties: { title },
+        parentPageId: superPage.entityId,
+      });
+    });
+
+    let subSubPage: PageFieldsFragment;
+    it("can create a subsubpage", async () => {
+      // - Super page 1
+      //   - sub page 1
+      //     - sub sub page 1
+      const title = "sub sub page 1";
+      subSubPage = await client.createPage({
+        accountId: existingUser.accountId,
+        properties: {
+          title,
+        },
+      });
+
+      const result = await client.setParentPage({
+        accountId: existingUser.accountId,
+        pageId: subSubPage.entityId,
+        parentPageId: subPage.entityId,
+      });
+
+      const pageTree = await client.getAccountPagesTree({
+        accountId: existingUser.accountId,
+      });
+
+      expect(result.entityId).toEqual(subSubPage.entityId);
+      expect(pageTree).toContainEqual({
+        entityId: subSubPage.entityId,
+        properties: { title },
+        parentPageId: subPage.entityId,
+      });
+    });
+
+    it("prevent multiple parents", async () => {
+      // - Super page 1
+      //   - sub page 1             <- because this page already has a parent
+      //     - sub sub page 1
+      //       - sub page 1 <- should not be allowed
+      await expect(
+        client.setParentPage({
+          accountId: existingUser.accountId,
+          pageId: subPage.entityId,
+          parentPageId: subSubPage.entityId,
+        }),
+      ).rejects.toThrowError(
+        /page with entityId = '.*' already has parent page entityId = '.*'/i,
+      );
+    });
```

Comment:
Have suggested elsewhere that we permit overwriting a page's parent, in which case we can still have this test but check that it's changed to a single parent.

Reply:
This is an interesting test case since re-parenting `sub page 1` under `sub sub page 1` would cause a cyclic dependency. I guess that's another case to solve